maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -627,9 +644,71 @@ private Model readModel( ModelSource modelSource, File pomFile, ModelBuildingReq
             throw problems.newModelBuildingException();
         }
 
+        if ( pomFile != null )
+        {
+            putCache( request.getModelCache(), modelSource, ModelCacheTag.FILEMODEL, model );
+        }
+
+        String groupId = getGroupId( model );
+        String artifactId = model.getArtifactId();
+        String version = getVersion( model );
+
+        ModelData modelData = new ModelData( modelSource, model, groupId, artifactId, version );
+        putCache( request.getModelCache(), groupId, artifactId, version, ModelCacheTag.RAW, modelData );
+
         return model;
     }
 
+    private Model getModelFromCache( ModelSource modelSource, ModelCache cache )
+    {
+        Model model;
+        if ( modelSource instanceof ArtifactModelSource )
+        {
+            ArtifactModelSource artifactModelSource = ( ArtifactModelSource ) modelSource;
+            ModelData modelData = getCache( cache, artifactModelSource.getGroupId(),
```

Comment:
The method name `getCache` suggests that it returns a cache. But one of its parameters is _named_ "cache" and it returns a `ModelData` instance. Does the name `getCache` match with what the method does?